scripts/rs/main.rs

```diff
@@ -0,0 +1,159 @@
+use std::process::Command;
+use std::thread::sleep;
+use std::time::Duration;
+use redis::{Commands, Connection, Client};
+use regex::Regex;
+use sysinfo::{System, Process};
+use chrono::Utc;
+use std::fs;
+use std::collections::HashMap;
+
+
+const REDIS_URL: &str = "redis://127.0.0.1/";
+const UPDATE_INTERVAL: u64 = 120; // 2 minutes
+
+
+fn run_command(cmd: &[&str]) -> Option<String> {
+    let output = Command::new(cmd[0]).args(&cmd[1..]).output().ok()?;
+    if output.status.success() {
+        Some(String::from_utf8_lossy(&output.stdout).to_string())
+    } else {
+        eprintln!("Error running command: {:?}", cmd);
+        None
+    }
+}
+
+fn convert_to_bytes(value: &str) -> u64 {
+    let re = Regex::new(r"(\d+\.?\d*)([a-zA-Z]+)").unwrap();
+    if let Some(caps) = re.captures(value) {
+        let num: f64 = caps[1].parse().unwrap_or(0.0);
+        let unit = &caps[2];
+        match unit.to_lowercase().as_str() {
+            "kb" => (num * 1024.0) as u64,
+            "mb" | "mib" => (num * 1024.0 * 1024.0) as u64,
+            "gb" | "gib" => (num * 1024.0 * 1024.0 * 1024.0) as u64,
+            _ => num as u64,
+        }
+    } else {
+        0
+    }
+}
+
+fn parse_docker_stats(output: &str) -> Vec<HashMap<String, String>> {
+    let lines: Vec<&str> = output.lines().collect();
+    if lines.len() < 2 { return vec![]; }
+
+    let keys: Vec<&str> = lines[0].split_whitespace().collect();
+    let mut stats_list = Vec::new();
+    
+    for line in &lines[1..] {
+        let values: Vec<&str> = line.split_whitespace().collect();
+        if values.len() >= keys.len() {
+            let mut stats = HashMap::new();
+            stats.insert("CONTAINER ID".to_string(), values[0].to_string());
+            stats.insert("NAME".to_string(), values[1].to_string());
+            stats.insert("CPU%".to_string(), values[2].trim_end_matches('%').to_string());
+            stats.insert("MEM_BYTES".to_string(), convert_to_bytes(values[3]).to_string());
+            stats.insert("MEM_LIMIT_BYTES".to_string(), convert_to_bytes(values[5]).to_string());
+            stats.insert("MEM%".to_string(), values[6].trim_end_matches('%').to_string());
+            stats.insert("NET_IN_BYTES".to_string(), convert_to_bytes(values[7]).to_string());
+            stats.insert("NET_OUT_BYTES".to_string(), convert_to_bytes(values[9]).to_string());
+            stats.insert("BLOCK_IN_BYTES".to_string(), convert_to_bytes(values[10]).to_string());
+            stats.insert("BLOCK_OUT_BYTES".to_string(), convert_to_bytes(values[12]).to_string());
+            stats.insert("PIDS".to_string(), values[13].to_string());
```

Comment:
this might be cleaner if the key was `&'static str`, no calls to .to_string() for keys. I also like to construct maps like `HashMap::from_iter([("CONTAINER_ID", values[0].to_string()), ("NAME", values[1].to_string()), ...]);` but both suggestions are just a matter of taste.